relay-server/src/metrics_extraction/transactions.rs

```diff
@@ -152,10 +168,73 @@ fn extract_user_satisfaction(
     None
 }
 
+/// These are the tags that are added to all extracted metrics.
+#[cfg(feature = "processing")]
+fn extract_universal_tags(
+    event: &Event,
+    custom_tags: &BTreeSet<String>,
+) -> BTreeMap<String, String> {
+    let mut tags = BTreeMap::new();
+    if let Some(release) = event.release.as_str() {
+        tags.insert("release".to_owned(), release.to_owned());
+    }
+    if let Some(dist) = extract_dist(event) {
+        tags.insert("dist".to_owned(), dist);
+    }
+    if let Some(environment) = event.environment.as_str() {
+        tags.insert("environment".to_owned(), environment.to_owned());
+    }
+    if let Some(transaction) = event.transaction.as_str() {
+        tags.insert("transaction".to_owned(), transaction.to_owned());
+    }
+
+    // The platform tag should not increase dimensionality in most cases, because most
+    // transactions are specific to one platform
+    let platform = event.platform.as_str().unwrap_or("other");
+    let platform = if store::is_valid_platform(platform) {
+        platform
+    } else {
+        "other"
+    };
```

Comment:
The way we emulate store normalization with `is_valid_platform` and 'normalize_dist` seems fragile and hard to maintain, but I don't have a better idea at the moment. Maybe run a `MetricsNormalizer` that is a subset of the current store normalizer?